languages/ruby/bitwarden_sdk/lib/bitwarden-sdk.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require 'json'
+require 'dry-types'
+
+require_relative 'schemas'
+require_relative 'extended_schemas/schemas'
+require_relative 'command_runner'
+require_relative 'bitwarden_lib'
+require_relative 'bitwarden_error'
+require_relative 'projects'
+require_relative 'secrets'
+
+module BitwardenSDK
+  class BitwardenSettings
+    attr_accessor :api_url, :identity_url
+
+    def initialize(api_url, identity_url)
+      # if api_url.nil? || identity_url.nil?
+      #   raise ArgumentError, "api_url and identity_url cannot be nil"
+      # end
+
+      @api_url = api_url
+      @identity_url = identity_url
+    end
+  end
+
+  class BitwardenClient
+    attr_reader :bitwarden, :project_client, :secrets_client
+
+    def initialize(bitwarden_settings)
+      client_settings = ClientSettings.new(
+        api_url: bitwarden_settings.api_url,
+        identity_url: bitwarden_settings.identity_url,
+        user_agent: 'Bitwarden RUBY-SDK',
+        device_type: nil
```

Comment:
```suggestion
        device_type: 'SDK'
```

Reply:
It worked for me fine when device_type is set to nil, however when I used 'SDK' I got this error: The client vault is locked and needs to be unlocked before use. Reverting to nil makes it work again. Do you know why I have this error?

Reply:
Hmm, yeah this might be something from the internal bitwarden_c library, for now we can leave this as it is then, thanks for looking into it.